### ISSUE TITLE
`Browser` object is not binding correctly for custom commands. 

### ISSUE THREAD

**Issue by nami-varthakavi** [webdriverio/webdriverio#3309]
The custom commands are not able to bind the browser object with them. Due to this inside the custom commands, we have to use `browser` operator instead of `this` operator to access any `browser` commands.

To provide an example : 

Test
 ```
describe('Verify navigation', () => {
    it('should navigate to the Home Page when clicked', () => {

        browser.url( 'http://webdriver.io');

        browser.forceClick(WDIOHomeWebPage.wdioLogo.selector);

        expect(browser.getTitle()).toBe(WDIOHomeWebPage.title);
    });
});
```

Page Object 
```
export default class WDIOHomeWebPage {
    get wdioLogo() { return $('body > div > div > header > a'); }
 }
    constructor() {
        this.title = 'WebdriverIO · Next-gen WebDriver test framework for Node.js';
    }
}
```

Custom Command 
```
const forceClickVar = function(selector) {
            const el = document.querySelector(selector);

            return el.click();
 };

browser.addCommand('forceClick', function(selector) {
         return  this.execute(forceClickVar, selector);
 });
``` 

In the above example the `add` Command throws an error `Error: Can't call execute on element with selector "undefined" because the element wasn't found` if we use ` this.execute(forceClickVar, element);`. 
The test passes only if we update the command to ` browser.execute(forceClickVar, element);`. 

This above scenario was also tested with other browser commands(`getWindow(), touchPerform(), pause()`) as well and they have the same behavior. 

Currently, there is a workaround for this to just use the `browser` object which is available globally through the test instead of the `this`.

**Comment by abjerstedt** [comment]
I spent some time on this, the following works as expected:
```
        test('browser and element should have different prototypes', async () => {
            const browser = await remote(remoteConfig)

            browser.addCommand('myCustomElementCommand', async function () {
                return this.execute(function () {return 1})
            })

            expect(await browser.myCustomElementCommand()).toBe(1);
        })
```
But if an element has been instantiated, the world melts:
```
        test('browser and element should have different prototypes', async () => {
            const browser = await remote(remoteConfig)
            const elem = await browser.$('#foo')

            browser.addCommand('myCustomElementCommand', async function () {
                return this.execute(function () {return 1})
            })

            expect(await browser.myCustomElementCommand()).toBe(1);
        })
```

**Comment by christian-bromann** [comment]
Fix is proposed in #3312
